frontend/model/contracts/group.js

```diff
@@ -519,6 +523,46 @@ sbp('chelonia/defineContract', {
             updateCurrentDistribution({ meta, state, getters })
           }
         }
+      },
+      sideEffect ({ meta, contractID, data }, { state, getters }) {
+        if (data.updatedProperties.status === PAYMENT_COMPLETED) {
+          const { loggedIn } = sbp('state/vuex/state')
+          const payment = state.payments[data.paymentHash]
+
+          if (loggedIn.username === payment.data.toUser) {
+            sbp('gi.notifications/emit', 'PAYMENT_RECEIVED', {
+              groupID: contractID,
+              creator: meta.username,
+              paymentHash: data.paymentHash,
+              amount: getters.withGroupCurrency(payment.data.amount)
+            })
+          }
+        }
+      }
+    },
+    'gi.contracts/group/sendPaymentThankYou': {
+      validate: objectOf({
+        fromUser: string,
+        toUser: string,
+        memo: string
+      }),
+      process ({ data }, { getters }) {
+        const thankYousFrom = getters.groupThankYousFrom
+
+        const fromUser = vueFetchInitKV(thankYousFrom, data.fromUser, {})
+        Vue.set(fromUser, data.toUser, data.memo)
```

Comment:
I can see that only one **thank you** could be saved which sent from `fromUser` to `toUser`. Is that by the requirement?

Reply:
Yep! From my previous review - we don't want to store all of the thank yous ever made, that's too much data in memory.